nuvlaedge/agent/agent.py

```diff
@@ -123,23 +123,44 @@ def __init__(self,
         # Report initial status
         NuvlaEdgeStatusHandler.starting(self.status_channel, 'agent')
 
-    @staticmethod
-    def check_uuid_missmatch(env_uuid, other_uuid):
-        """ This method checks if the UUID provided as an environmental variable matches the one stored locally
-        Args:
-            env_uuid: UUID parsed as an environmental variable
-            other_uuid: UUID stored locally or retrieved from API keys
-
-        Raises: AgentSettingsMissMatch if the UUID's are not equal
+    def check_uuid_missmatch(self):
+        """ Check if the UUID provided in the settings is the same as the one stored in the local session.
+        Check for possible settings missmatch
+        Check here for settings missmatch. The priority comes as follows:
+        1. Arg1 == Arg2 : All good, continue
+        2. Arg1 != Arg2 : Update Arg1 with Arg2
+        3. Arg1 == None : Update Arg1 with Arg2
+        4. Arg2 == None : New NE installation, no need to update
 
         """
+        def sanitise_uuid(uuid: NuvlaID) -> tuple[bool, str]:
+            if uuid is None:
+                return False, ""
+            return uuid.validate(), uuid.split("/")[-1]
+
+        env_valid, env_uuid = sanitise_uuid(self.settings.nuvlaedge_uuid)
+        local_valid, local_uuid = sanitise_uuid(self._nuvla_client.nuvlaedge_uuid)
```

Comment:
AFAIK `uuid.validate()` will return `False` if the value is, for example, `69a0b1aa-f27f-11ee-8927-8f4e8dcf9e99`